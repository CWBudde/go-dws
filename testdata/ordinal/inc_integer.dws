// Test Inc() with integers
// Task 9.24: Inc() built-in function

var x: Integer := 5;
PrintLn('Initial value: ', x);

Inc(x);
PrintLn('After Inc(x): ', x);

Inc(x, 3);
PrintLn('After Inc(x, 3): ', x);

var delta: Integer := 10;
Inc(x, delta);
PrintLn('After Inc(x, delta) where delta=10: ', x);

// Test with negative values
var y: Integer := -5;
PrintLn('Negative initial: ', y);
Inc(y);
PrintLn('After Inc(y): ', y);
Inc(y, 10);
PrintLn('After Inc(y, 10): ', y);
