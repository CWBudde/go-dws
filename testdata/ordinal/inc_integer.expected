Initial value:  5
After Inc(x):  6
After Inc(x, 3):  9
After Inc(x, delta) where delta=10:  19
Negative initial:  -5
After Inc(y):  -4
After Inc(y, 10):  6
